### PR TITLE
initializing agent area dictionary correctly when None is provided. 

### LINKED ISSUE #121: SwitchAreaAgent.switch_area.initialize_switch_area() fails when switch_area_dict is None.
**Describe the bug**
If an Instance SwitchAreaAgent is created with switch_area_dict=None then the call to self.switch_area.initialize_switch_area(self.agent_area_dict) fails with the following error:
 
Traceback (most recent call last):
  File "/home/vale/git/gridappsd-distributed-static-ybus-service/dsys/distributed_static_ybus_service.py", line 358, in <module>
    main(system_bus_config_file=systemBusConfigFile, 
  File "/home/vale/git/gridappsd-distributed-static-ybus-service/dsys/distributed_static_ybus_service.py", line 275, in main
    switchAreaService.connect()
  **File "/home/vale/git/gridappsd-distributed-static-ybus-service/.venv/lib/python3.10/site-packages/gridappsd/field_interface/agents/agents.py", line 273, in connect
    self.switch_area.initialize_switch_area(self.agent_area_dict)
  File "/home/vale/git/gridappsd-distributed-static-ybus-service/.venv/lib/python3.10/site-packages/cimgraph/models/switch_area.py", line 52, in initialize_switch_area
    for sec_area_msg in switch_msg['secondary_areas']:**

**To Reproduce**
Create an instance of SwitchAreaAgent with the argument switch_area_dict=None then call the connect function of that agent.

The error is due to self.agent_area_dict returned by the context manager for this switch area somehow returned the feeder area context dictionary. The code for initialize_switch_area() in switch_area.py is expecting the switch area context dictionary.

This behavior will certainly cause a similar issue when creating an instance of SecondaryAreaAgent with secondary_area_dict=None as well.

### PR DESCRIPTION
**Pull request by poorva1209**
Updated message bus connection to occur at agent initialization 
Fixes: #121

### CODE REVIEW

File: gridappsd-field-bus-lib/gridappsd/field_interface/agents/agents.py [hunk 1]
```diff
@@ -87,7 +91,7 @@ def __init__(self,
         #    self.addressable_equipments = agent_dict['addressable_equipment']
         #    self.unaddressable_equipments = agent_dict['unaddressable_equipment']
 
-    def connect(self):
+    def _connect(self):
```
Comment (craig8) [inline gridappsd-field-bus-lib/gridappsd/field_interface/agents/agents.py:94]: Why are we using _ here?

File: gridappsd-field-bus-lib/gridappsd/field_interface/agents/agents.py [hunk 2]
```diff
@@ -109,6 +113,13 @@ def connect(self):
         if ('context_manager' not in self.app_id):
             LocalContext.register_agent(self.downstream_message_bus,
                                         self.upstream_message_bus, self)
+            
+    def disconnect(self):
```
Comment (craig8) [inline gridappsd-field-bus-lib/gridappsd/field_interface/agents/agents.py:117]: But we aren't using _ here?

File: gridappsd-field-bus-lib/gridappsd/field_interface/agents/agents.py [hunk 3]
```diff
@@ -265,17 +271,13 @@ def __init__(self,
...
 
-    def connect(self):
-        super().connect()
-        if self.switch_area is None:
+        self._connect()
+
+        if self.agent_area_dict is not None:
             self.switch_area = SwitchArea(self.downstream_message_bus_def.id,
                                           self.connection)
+            print(self.agent_area_dict)
```
Comment (craig8) [inline gridappsd-field-bus-lib/gridappsd/field_interface/agents/agents.py:280]: Remove this or use log with some sort of label

File: gridappsd-field-bus-lib/gridappsd/field_interface/context.py [hunk 4]
```diff
@@ -13,6 +13,8 @@ def get_context_by_feeder(cls, downstream_message_bus: FieldMessageBus, feeder_m
         request = {'request_type' : 'get_context',
                     'modelId': feeder_mrid,
                    'areaId': area_id}
+        print("*****************************")
```
Comment (craig8) [inline gridappsd-field-bus-lib/gridappsd/field_interface/context.py:16]: Remove print make it a log. message

**Review by craig8 — CHANGES_REQUESTED** [review]

File: gridappsd-field-bus-lib/gridappsd/field_interface/agents/agents.py (hunk 1 above)
Comment (poorva1209) [reply gridappsd-field-bus-lib/gridappsd/field_interface/agents/agents.py:94]: It is used internally within this class while initialization. Not called by any agent.

File: gridappsd-field-bus-lib/gridappsd/field_interface/agents/agents.py (hunk 2 above)
Comment (poorva1209) [reply gridappsd-field-bus-lib/gridappsd/field_interface/agents/agents.py:117]: yeah, agent uses intialization to connect but can disconnect is some properties in topology is not applicable to their app like not having any addressable equipment. Having initialization and connect function and spawn function was causing more confusion before.

**Review by craig8 — APPROVED** [review]

**Pull request merged**